app/javascript/gobierto_dashboards/modules/contracts_controller.js

```diff
@@ -91,6 +91,14 @@ export class ContractsController {
         EventBus.$on('summary_ready', () => {
           this._renderSummary();
         });
+
+        EventBus.$on('filter_changed', (options) => {
+          this._updateChartsFromFilter(options);
+        });
+
+        EventBus.$once('moved_to_summary', () => {
+          this._redrawCharts();
+        });
```

Comment:
Try to include an explanatory comment for each event. Specially the events. You may do it in following PRs

Reply:
I'll add an explanation for that.